### DISCUSSION TITLE
Android de çalıştırılabilir mi?

### DISCUSSION THREAD

**Discussion by ghost** [KebabLord/turkanime-indirici#21] [Genel] [answered]
Yani normalde çalısması gerekir ama sorayım dedim

**Comment by KebabLord** [comment]
Tek sıkıntı selenium.
![2021-06-12_13-28](https://user-images.githubusercontent.com/24779257/121773065-1e54e800-cb82-11eb-8fee-50eb94a119c6.png)

Geckodriver pek çok android sürümün işlemci mimarisi olan arm7hf için release yayınlamayı [v0.23.0](https://github.com/mozilla/geckodriver/releases/tag/v0.23.0) sürümünden itibaren bıraktı. Aslında direk bu sürüm kullanılabilir ancak Termux reposunda var mı hatırlamıyorum ama cihazda ilaveten headless çalışmak üzere bir firefox tarayıcısı (desktop için hazırlanmış) sürüm kurulmalı. Belirteyim ki mozilla'nın yayınladığı geckodriver binary'sini kendi arm7hf cihazımda çalıştıramamıştım bile.

Lakin önemli bir gelişmedir ki, [v0.26.0](https://github.com/mozilla/geckodriver/releases/tag/v0.26.0) sürümünden itibaren geckodriver android firefox ile çalışma desteği kazandı. Yani uzun lafın kısası teorik olarak, şu şekilde android'de bu scripti çalıştırmak mümkün:

1.  Firefox android'i telefona indirmek
2.  Son geckodriver sürümünü telefon içinde derlemek
3.  Termux kullanarak scripti çalıştırmak

İlaveten hatırlatmalıyım ki, script aarch64 mimariye sahip olan raspberry pi4 üstünde başarıyla çalışıyor (#16).
Yani ARM işlemciler üstünde bu scripti kullanmak yine mümkün.
![image](https://user-images.githubusercontent.com/24779257/227743717-4223098b-51f4-4633-87bd-a6a01b9b6c71.png)

**Comment by KebabLord** [comment, marked as answer]
Script artık selenium'dan kurtulduğundan, ve curl-cffi'nin beta sürümüne android desteği geldiğinden, script artık android'de de sorunsuz çalıştırılabiliyor
https://github.com/KebabLord/turkanime-indirici/wiki/Android'de-%C3%A7al%C4%B1%C5%9Ft%C4%B1rmak
